File: Games/Dennis/src/de/hsh/dennis/model/GameModelUtils.java
package de.hsh.dennis.model;

import de.hsh.dennis.model.NpcLogic.NpcHandler;
import de.hsh.dennis.model.audio.AudioConfig.DelayBetweenSpawns;
import de.hsh.dennis.model.audio.AudioConfig.MovingSpeeds;
import de.hsh.dennis.model.audio.AudioConfig.Mp3Names;

final class GameModelUtils {

    private GameModelUtils() {}

    static void initEasyDifficulty( NpcHandler npcHandler ) {
        npcHandler.getAudioAnalyzer().setSensitivity( 0.2d );
        npcHandler.setDelaysBetweenSpawns( DelayBetweenSpawns._easy );
        npcHandler.generateNpcs( Mp3Names.easy, MovingSpeeds._easy );
        AudioPlayer.MusicPlayer.loadFile( Mp3Names.easy );
    }

    static void initMediumDifficulty( NpcHandler npcHandler ) {
        npcHandler.getAudioAnalyzer().setSensitivity( 0.1d );
        npcHandler.setDelaysBetweenSpawns( DelayBetweenSpawns._medium );
        npcHandler.generateNpcs( Mp3Names.medium, MovingSpeeds._medium );
        AudioPlayer.MusicPlayer.loadFile( Mp3Names.medium );
    }

    static void initHardDifficulty( NpcHandler npcHandler ) {
        npcHandler.getAudioAnalyzer().setSensitivity( 0.05d );
        npcHandler.setDelaysBetweenSpawns( DelayBetweenSpawns._hard );
        npcHandler.generateNpcs( Mp3Names.hard, MovingSpeeds._hard );
        AudioPlayer.MusicPlayer.loadFile( Mp3Names.hard );
    }

    static void initNightmareDifficulty( NpcHandler npcHandler ) {
        npcHandler.getAudioAnalyzer().setSensitivity( 0.0d );
        npcHandler.setDelaysBetweenSpawns( DelayBetweenSpawns._nightmare );
        npcHandler.generateNpcs( Mp3Names.nightmare, MovingSpeeds._nightmare );
        AudioPlayer.MusicPlayer.loadFile( Mp3Names.nightmare );
    }
}

File: Games/Dennis/src/de/hsh/dennis/model/GameModelUtils.kt
package de.hsh.dennis.model

import de.hsh.dennis.model.Spawn.NpcHandler
import de.hsh.dennis.model.audio.AudioAnalyser
import de.hsh.dennis.model.audio.AudioConfig.*

internal object GameModelUtils {

    private fun NpcHandler.initDifficulty(
            sensitivity: Double,
            spawnDelay: Double,
            audioFileName: String,
            movingSpeed: Double
    ): Double {
        AudioAnalyser.sensitivity = sensitivity
        AudioAnalyser.spawnDelay = spawnDelay
        this.generateNpcs(audioFileName, movingSpeed)
        MusicPlayer.loadFile(audioFileName)
        return movingSpeed
    }

    fun initEasyDifficulty(npcHandler: NpcHandler): Double  = npcHandler.initDifficulty(
            0.2,
            DelayBetweenSpawns._easy,
            Mp3Names.easy,
            MovingSpeeds._easy
    )

    fun initMediumDifficulty(npcHandler: NpcHandler): Double  = npcHandler.initDifficulty(
            0.1,
            DelayBetweenSpawns._medium,
            Mp3Names.medium,
            MovingSpeeds._medium
    )

    fun initHardDifficulty(npcHandler: NpcHandler): Double  = npcHandler.initDifficulty(
            0.05,
            DelayBetweenSpawns._hard,
            Mp3Names.hard,
            MovingSpeeds._hard
    )

    fun initNightmareDifficulty(npcHandler: NpcHandler): Double  = npcHandler.initDifficulty(
            0.01,
            DelayBetweenSpawns._nightmare,
            Mp3Names.nightmare,
            MovingSpeeds._nightmare
    )
}
